src/core/ipe-setup.c

```diff
@@ -0,0 +1,127 @@
+         * will show up as a directory under /sys/kernel/security/ipe/policies/ and there will be an
+         * 'active' file under it to write into, which will make it active. */
+        for (;;) {
+                _cleanup_free_ char *line = NULL;
+                char *p;
+
+                r = read_line(finput, LONG_LINE_MAX, &line);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to read the IPE policy "IPE_POLICY_PATH": %m");
+                if (r == 0)
+                        break;
+
+                p = strstrafter(line, "policy_name=");
+                if (p && !policy_name) {
+                        policy_name = strdup(p);
+                        if (!policy_name)
+                                return log_oom();
+                        p = strchr(policy_name, ' ');
+                        if (p)
+                                *p = '\0';
+                }
+
+                p = strstrafter(line, "policy_version=");
+                if (p && !policy_version) {
+                        policy_version = strdup(p);
+                        if (!policy_version)
+                                return log_oom();
+                        p = strchr(policy_version, ' ');
+                        if (p)
+                                *p = '\0';
+                }
+
+                if (policy_name && policy_version)
+                        break;
+        }
+
+        if (!policy_name || !policy_version)
+                return log_error_errno(ENOENT, "Failed to find policy_name and policy_version in the IPE policy "IPE_POLICY_PATH);
+
+        policy_existing_version_file = strjoin(IPE_SECFS_POLICIES, policy_name, "/version");
+        if (!policy_existing_version_file)
+                return log_oom();
+
+        r = read_virtual_file(policy_existing_version_file, SIZE_MAX, &policy_existing_version, /* ret_size= */ NULL);
+        if (r < 0 && r != -ENOENT)
+                return log_error_errno(r, "Failed to read the existing IPE policy version: %m");
+        if (r >= 0) {
+                if (strverscmp_improved(policy_version, policy_existing_version) <= 0) {
+                        log_info("The existing IPE policy '%s' is at version '%s', the new version is '%s', skipping.",
+                                 policy_name,
+                                 policy_existing_version,
+                                 policy_version);
+                        return 0;
+                }
+
+                /* Policy is already installed and older, update it */
+                output_path_free = strjoin(IPE_SECFS_POLICIES, policy_name, "/update");
+                if (!output_path_free)
+                        return log_oom();
+                output_path = output_path_free;
+        } else {
+                /* Policy is not installed, install it and activate it */
+                output_path = IPE_SECFS_NEW_POLICY;
+        }
```

Comment:
no {} around single-line if blocks

Reply:
...there's a comment, surely that counts? ;-)